buildSrc/src/main/kotlin/otel.muzzle-check.gradle.kts

```
@@ -0,0 +1,351 @@
+  if (versionArtifact != null) {
+    val dep = (dependencies.create(versionArtifact.run { "${groupId}:${artifactId}:${version}" }) as ModuleDependency).apply {
+      isTransitive = true
+      exclude("com.sun.jdmk", "jmxtools")
+      exclude("com.sun.jmx", "jmxri")
+      for (excluded in muzzleDirective.excludedDependencies.get()) {
+        val (group, module) = excluded.split(':')
+        exclude(group, module)
+      }
+    }
+    config.dependencies.add(dep)
+
+    for (additionalDependency in muzzleDirective.additionalDependencies.get()) {
+      val additional = if (additionalDependency.count { it == ':' } < 2) {
+        // Dependency definition without version, use the artifact's version.
+        "${additionalDependency}:${versionArtifact.version}"
+      } else {
+        additionalDependency
+      }
+      val additionalDep = (dependencies.create(additional) as ModuleDependency).apply {
+        isTransitive = true
+      }
+      config.dependencies.add(additionalDep)
+    }
+  }
+
+  val muzzleTask = tasks.register(taskName) {
+    dependsOn(configurations.named("runtimeClasspath"))
+    doLast {
+      val instrumentationCL = createInstrumentationClassloader()
+      val ccl = Thread.currentThread().contextClassLoader
+      val bogusLoader = BogusClassLoader()
+      val userCL = createClassLoaderForTask(config)
+      Thread.currentThread().contextClassLoader = bogusLoader
+      try {
+        // find all instrumenters, get muzzle, and assert
+        val assertionMethod = instrumentationCL
+          .loadClass("io.opentelemetry.javaagent.tooling.muzzle.matcher.MuzzleGradlePluginUtil")
+          .getMethod(
+            "assertInstrumentationMuzzled",
+            ClassLoader::class.java,
+            ClassLoader::class.java,
+            Boolean::class.javaPrimitiveType)
+        assertionMethod.invoke(
+          null,
+          instrumentationCL,
+          userCL,
+          muzzleDirective.assertPass.get())
+      } finally {
+        Thread.currentThread().contextClassLoader = ccl
+      }
+
+      for (thread in Thread.getAllStackTraces().keys) {
+        if (thread.contextClassLoader === bogusLoader
+          || thread.contextClassLoader === instrumentationCL
+          || thread.contextClassLoader === userCL) {
+          throw GradleException(
+            "Task ${taskName} has spawned a thread: ${thread} with classloader ${thread.contextClassLoader}. " +
+              "This will prevent GC of dynamic muzzle classes. Aborting muzzle run.")
+        }
+      }
+    }
+  }
+
+  runAfter.configure { finalizedBy(muzzleTask) }
+  return muzzleTask
+}
+
+fun createClassLoaderForTask(muzzleTaskConfiguration: Configuration): ClassLoader {
+  val userUrls = objects.fileCollection()
+  logger.info("Creating task classpath")
+  userUrls.from(muzzleTaskConfiguration.resolvedConfiguration.files)
+  return classpathLoader(userUrls.plus(muzzleTooling), ClassLoader.getPlatformClassLoader())
```

Shouldn't that be the bootstrap conf?
```suggestion
  return classpathLoader(userUrls.plus(muzzleBootstrap), ClassLoader.getPlatformClassLoader())
```

---

Oops - guess it still passed since bootstrap gets included transitively